econml/cate_estimator.py

```diff
@@ -8,9 +8,13 @@
 from functools import wraps
 from copy import deepcopy
 from warnings import warn
+from collections import defaultdict
+import shap
```

Comment:
We should create a single python file shap.py where all the interaction with shap happens. shap should only be imported in that file and nowhere else. All functionality that directly calls shap, should be happening in that file and the functions in any other files should be calling the intermediate functionalities we want.

This way we create a single interaction point and if shap changes anything in the future all we would need to do is change our shap.py file to handle the changes and offer the same api to the remainder of the code.

Also this would allow our users to not have any breaking changes unless they wont to use shap.